src/OpenSSL/crypto.py

```diff
@@ -1735,14 +1736,13 @@ def dump_privatekey(type, pkey, cipher=None, passphrase=None):
     elif type == FILETYPE_TEXT:
         rsa = _lib.EVP_PKEY_get1_RSA(pkey._pkey)
         result_code = _lib.RSA_print(bio, rsa, 0)
-        # TODO RSA_free(rsa)?
+        _ffi.gc(rsa, _lib.RSA_free)
```

Comment:
This should be `rsa = _ffi.gc(rsa, _lib.RSA_free)` and it should be on the line right below `rsa = _lib.EVP_PKEY...`